examples/server/server.cpp

```diff
@@ -676,6 +676,28 @@ struct llama_server_context
             }
         }
 
+        const auto &samplers_sequence = data.find("samplers");
+        if (samplers_sequence != data.end() && samplers_sequence->is_array())
+        {
+            std::vector<std::string> sampler_names;
+            for (const auto &sampler_name : *samplers_sequence)
+            {
+                if (sampler_name.is_string())
+                {
+                    sampler_names.emplace_back(sampler_name);
+                }
+            }
+            auto sampler_types = sampler_types_from_names(sampler_names, false);
+            slot->sparams.samplers_sequence.swap(sampler_types);
+        }
+        else
+        {
+            // cloning the default samplers into the current params,
+            // while keeping sparams.samplers_sequence reference intact
+            auto default_sampler_types = default_sparams.samplers_sequence;
+            slot->sparams.samplers_sequence.swap(default_sampler_types);
```

Comment:
How is this different from:

```suggestion
            slot->sparams.samplers_sequence = default_sampler_types;
```

Reply:
Yeah, I got a little bit confused here. Uploaded the fixed version (the "suggested change" won't compile).